commit dba14c58c3edd3b45ba4c35522ec89ea785f7df0
Author: Volker Krause <vkrause@kde.org>
Date:   Tue Jan 2 18:11:03 2024 +0100

    Fix opening temporary content: URIs from application content providers
    
    e5d591a0d09 focused on document or tree URIs you'd get from the file
    dialog, but we can also get ones via Intents from application-specific
    content providers (e.g. opening an attachment from an email client).
    Those need to be handled like a single document here, parsing them
    like a tree URI results in a bunch of Java exceptions down the line.
    
    Change-Id: I19cb1075fd6a3ccad927543f8bc5a63566f4877e
    Pick-to: 6.7 6.6 6.5

diff --git a/src/plugins/platforms/android/androidcontentfileengine.cpp b/src/plugins/platforms/android/androidcontentfileengine.cpp
index cd7913fddf..52ca0114c7 100644
--- a/src/plugins/platforms/android/androidcontentfileengine.cpp
+++ b/src/plugins/platforms/android/androidcontentfileengine.cpp
@@ -608,7 +608,7 @@ DocumentFilePtr DocumentFile::parseFromAnyUri(const QString &fileName)
     const QString encodedUri = QUrl(fileName).toEncoded();
     const QJniObject uri = parseUri(encodedUri);
 
-    if (DocumentsContract::isDocumentUri(uri))
+    if (DocumentsContract::isDocumentUri(uri) || !DocumentsContract::isTreeUri(uri))
         return fromSingleUri(uri);
 
     const QString documentType = "/document/"_L1;
